plutus-ir/src/Language/PlutusIR/Compiler/Types.hs

```diff
@@ -85,6 +86,7 @@ type Compiling m e uni a =
     , AsError e uni (Provenance a)
     , MonadError e m
     , MonadQuote m
+    , uni `PLC.Includes` ()
```

Comment:
This one I also think should maybe go in favour of using the scott-encoded unit. It's nice to be able to use PIR over absolutely any universe!

Reply:
Do you share the definition of `unit` in the IR compiler? If it was

```haskell
TyForall () a (Type ()) . TyFun () (TyVar () a) $ TyVar () a
```

everywhere, then the new

```haskell
TyBuiltin () (Some (TypeIn unit))
```

is an optimization size-, cost- and readability-wise. I'm not too keen on undoing an optimization, I think it's fine to ask the user to include `()` into their universe.

Reply:
Actually, no, you're right.

Reply:
I agree it's smaller and more readable... I just think going from "any universe" to "restricted universe" is a much bigger step than "restricted universe" to "more restricted universe". At the moment the PIR compiler is totally unrestricted!

Reply:
It's more like "going from any *particular* universe to a *particular* universe with `()`". The caller has to specify the universe, so I don't think it's a big deal to ask the caller to add `()` to that universe as they most likely need to deal with other built-in types like `Integer` anyway.

Anyway, I'll rollback the changes just so that we don't need to argue about them.

Reply:
I think it's somewhat unreasonable. For example, we only need `()` to deal with `trace`, so if we didn't have that (which we may not have on-chain), then this would force us to add a redundant builtin type.

Reply:
Let's just agree that we're both wrong. Now that we don't use that "efficiency is not important" argument, implementing `delay` & `force` in terms of `unit` doesn't sound appealing regardless of how `unit` is implemented (unshared Scott-encoded `unit` is clearly wasteful, though). @kwxm had some plans on adding `delay` & `force` as primitives.